CMakeLists.txt

```diff
@@ -344,7 +344,9 @@ if(VLLM_GPU_LANG STREQUAL "CUDA")
        "csrc/quantization/cutlass_w8a8/c3x/scaled_mm_sm90_fp8.cu"
        "csrc/quantization/cutlass_w8a8/c3x/scaled_mm_sm90_int8.cu"
        "csrc/quantization/cutlass_w8a8/c3x/scaled_mm_azp_sm90_int8.cu"
-       "csrc/quantization/cutlass_w8a8/c3x/scaled_mm_blockwise_sm90_fp8.cu")
+       "csrc/quantization/cutlass_w8a8/c3x/scaled_mm_blockwise_sm90_fp8.cu"
+       "csrc/quantization/cutlass_w8a8/grouped_mm_c3x.cu"
+       "csrc/quantization/cutlass_w8a8/moe_data.cu")
```

Comment:
These need to go in their own block that's compiled for CUDA 12.3 or later.

For example see what we do for the 2:4 sparse kernels to compile on 12.2 or later https://github.com/vllm-project/vllm/blob/27df5199d99627e1eb101071c2155f888181bd64/CMakeLists.txt#L424-L444